app/upgrades.go

```diff
@@ -76,6 +76,35 @@ var handlers = map[string]appUpgrade{
 	"desert":  {},
 	"eigengrau": {
 		Handler: func(app *App, ctx sdk.Context, plan upgradetypes.Plan) (module.VersionMap, error) {
+			nhashName := "Hash"
+			nhashSymbol := "HASH"
+			nhash, found := app.BankKeeper.GetDenomMetaData(ctx, "nhash")
+			if found {
```

Comment:
If you have the set all option... why have the set partial option as well?

Reply:
I wanted to handle if it's not found. I thought about returning an error, but then I figured it was just as easy to just create it if it's not found. I like the get, update, set way better because then we're not undoing anything that might possibly have been changed since it was created. But if it hasn't been created yet, we know we're not undoing anything. And even though I know that there's very very little chance that it has been changed from the original `bluetiful`, I'm just erring on the side of caution.